test/Choose.hs

```diff
@@ -20,23 +20,23 @@ tests = testGroup "Choose"
 
 
 gen
-  :: (Has Choose sig m, Show a)
-  => (forall a . Show a => Gen a -> Gen (With (m a)))
+  :: Has Choose sig m
+  => (forall a . Gen a -> Gen (m a))
   -> Gen a
-  -> Gen (With (m a))
-gen m a = subterm2 (m a) (m a) (liftWith2 "(<|>)" (<|>))
+  -> Gen (m a)
+gen m a = addLabel "<|>" (infixL 3 "<|>" (<|>) <*> m a <*> m a)
```

Comment:
There’s no automatically labelling infix constructor—there are only two places we want to label infix operators—so we use `addLabel` to automatically classify the term.